api/v1alpha3/awsmachine_types.go

```diff
@@ -70,8 +70,15 @@ type AWSMachineSpec struct {
 	// +optional
 	AdditionalSecurityGroups []AWSResourceReference `json:"additionalSecurityGroups,omitempty"`
 
+	// FailureDomain is the failure domain unique identifier this Machine should be attached to, as defined in Cluster API.
+	// For this infrastructure provider, the ID is equivalent to an AWS Availability Zone.
+	// If multiple subnets are matched for the availability zone, the first one return is picked.
+	FailureDomain *string `json:"failureDomainID,omitempty"`
```

Comment:
If we are going generic with the naming here, should we add it to the Machine resource instead?

Reply:
That's another solution, in that case we'd lose the comment and description specific to the infrastructure provider though